src/notation/view/abstractelementpopupmodel.cpp

```diff
@@ -212,7 +214,17 @@ const mu::engraving::ElementTypeSet& AbstractElementPopupModel::dependentElement
 
 void AbstractElementPopupModel::updateItemRect()
 {
-    QRect rect = m_item ? fromLogical(m_item->canvasBoundingRect()).toQRect() : QRect();
+    QRect rect;
+
+    if (!m_item) {
+        rect = QRect();
+    } else {
+        if (m_item->isDynamic()) {
+            rect = fromLogical(toDynamic(m_item)->adjustedBoundingRect()).toQRect();
```

Comment:
Sorry, same thing again...

Reply:
It is used to place the popup below the `adjustedBoundingRectangle` of dynamic like you can see in the in the below image, even when we zoom in it is placed consistently below the rectangle.
<img width="200px" src="https://github.com/user-attachments/assets/cafe62ba-2c92-44e4-9cc2-14cbbe84cb44" />
If we use the `canvasBoundingRect` the popup will be inside the bounding rectangle on zooming (even if I add some value to root.y of updatePosition in Qml file it was not getting placed consistently on zooming, so thats why I used `adjustedBoudingRect` instead of `canvasBoundingRect` as the `elementRect`)
<img width="200px" src="https://github.com/user-attachments/assets/433d1abb-efcc-4663-9042-3dee71d4359e" />

Reply:
I see... @mike-spa do you have ideas about how this can be generalised? A bit like `hitShape()`, but different...

Also, since all TextBase elements have this grey box around them while editing their text, it might make sense to apply this to all of them. But only during actual text editing.